docs/source/developer/contributing.rst

```diff
@@ -422,6 +439,145 @@ as to free up core developer time. Examples of ways to help:
+    * [ ] The patch is implemented using the high-level extension API only
+          (https://numba.readthedocs.io/en/stable/extending/high-level.html).
+    * [ ] The patch updates the documentation to reflect the new functionality.
+          (https://github.com/numba/numba/blob/main/docs/source/reference/numpysupported.rst)
+    * [ ] The patch is line wrapped to 80 characters.
+    * [ ] Changes to Python source in the patch are `flake8` compliant.
+    * [ ] The implementation of the NumPy function reflects that present in the
+        NumPy implementation itself (look in the NumPy repository:
+          https://github.com/numpy/numpy)...
+        * [ ] The implementation has references to the region of the NumPy code
+              base that provides the same functionality. The references are URLs
+              of the form:
+              `https://github.com/numpy/numpy/tree/<SHA>/path/to/file.py:#<lines>`
+        * [ ] The code "looks" reasonably similar to the NumPy implementation.
+        * [ ] The code uses the same algorithm as the NumPy implementation, if
+              it does not it explains why and this justification is reasonable.
+        * [ ] The type inference part of the Numba implementation restricts the
+              types to those declared in the NumPy docs (or if not present, what
+              is accepted in the NumPy implementation itself).
+        * [ ] The Numba implementation uses the same variable names for
+              arguments as in the NumPy implementation.
+        * [ ] The Numba code has commentary present for anything not-obvious by
+              inspection.
+        * [ ] The Numba code accommodates differences that might occur across
+              different NumPy versions in a clear fashion and uses
+              `numba.np.numpy_support.numpy_version` to separate the difference
+              in implementations.
+    * [ ] The unit tests for the implementation:
+        * [ ] Use the same test inputs as the NumPy unit tests for the same
+              functionality. If this is not the case there are comments which
+              explain why and this justification is reasonable.
+        * [ ] The NumPy unit test inputs have reference URLs associated with
+              them to track their origin. The references are URLs of the form:
+              `https://github.com/numpy/numpy/tree/<SHA>/path/to/file.py:#<lines>`
+        * [ ] Use `unittest` style assertions (not pytest style!).
+        * [ ] Check that unsupported/invalid input types are handled as expected
+              using the `with self.assertRaises()` pattern.
+        * [ ] Explicitly check custom error messages are raised appropriately.
+        * [ ] Hit all the branches of the proprosed change (both at type
+              inference and run time).
+        * [ ] Check that numerically awkward values work as expected e.g.
+              `NaN`, `+/-Inf`, `0`, `INT_MAX`, `INT_MIN` etc.
```

Comment:
I wonder if we should reference Boundary Value Testing in addition to this as this is a subset of BVT. BVT will also include empty containers.